lookup.go

```diff
@@ -140,12 +158,56 @@ func doInput(in chan<- string, path string, wg *sync.WaitGroup) error {
 			log.Fatal("unable to open output file:", err.Error())
 		}
 	}
-	s := bufio.NewScanner(f)
-	for s.Scan() {
-		in <- s.Text()
-	}
-	if err := s.Err(); err != nil {
-		log.Fatal("input unable to read file", err)
+	//TK caching
+	if moduleType == "ZONE" {
+		tokens := dns.ParseZone(f, ".", path)
+		cached := TargetedDomain{"", []string{}}
+		for t := range tokens {
+			if t.Error != nil {
+				continue
+			}
+			switch record := t.RR.(type) {
+			case *dns.NS:
+				if strings.Count(t.RR.Header().Name, ".") < 2 {
+					continue
+				}
+				name := t.RR.Header().Name
+				if name[len(name)-1] == '.' {
+					name = name[0 : len(name)-1]
+				}
+				if cached.Domain == "" {
+					cached = TargetedDomain{name, []string{record.Ns}}
+					continue
+				}
+				if name == cached.Domain {
+					cached.Nameservers = append(cached.Nameservers, record.Ns)
+					continue
+				}
+				query, err := json.Marshal(cached)
+				cached = TargetedDomain{name, []string{record.Ns}}
+				if err != nil {
+					log.Fatal("Internal Error: %s", err)
+				}
+				in <- string(query)
+			default:
+				continue
+			}
+		}
+		if cached.Domain != "" {
+			query, err := json.Marshal(cached)
+			if err != nil {
+				log.Fatal("Internal Error: %s", err)
+			}
+			in <- string(query)
+		}
+	} else {
+		s := bufio.NewScanner(f)
+		for s.Scan() {
+			in <- s.Text()
+		}
+		if err := s.Err(); err != nil {
+			log.Fatal("input unable to read file", err)
+		}
```

Comment:
Same question as earlier---why is this in the framework and not in the zone module itself?